Firmware/communication/can/can_simple.cpp

```diff
@@ -269,7 +269,20 @@ void CANSimple::set_input_torque_callback(Axis& axis, const can_Message_t& msg)
 }
 
 void CANSimple::set_controller_modes_callback(Axis& axis, const can_Message_t& msg) {
-    axis.controller_.config_.control_mode = static_cast<Controller::ControlMode>(can_getSignal<int32_t>(msg, 0, 32, true));
+
+    Controller::ControlMode const mode = static_cast<Controller::ControlMode>(can_getSignal<int32_t>(msg, 0, 32, true));
+    if (mode == Controller::CONTROL_MODE_POSITION_CONTROL) {
+        float estimate = 0.0f;
+        if (axis.controller_.config_.circular_setpoints) {
+            estimate = axis.encoder_.pos_circular_.any().value_or(0.0f);
+        } else {
+            estimate = axis.encoder_.pos_estimate_.any().value_or(0.0f);
+        }
+
+        axis.controller_.set_input_pos_and_steps(estimate);
+    }
```

Comment:
This reaction on writing the control mode is appropriate, but we need the behavior to be consistent when set from other sources than CAN. My suggestion is:
1. Merge to a reusable function this snippet with the "// To avoid any transient on startup" section in `axis.cpp start_closed_loop_control`. You could call it `set_setpoint_to_estimate`.
2. Add a `c_setter` tag to the `control_mode` endpoint in `odrive-interface.yaml`, so this can get executed when we write it on USB or UART.
3. Add an entry in the changelog about this reaction on control mode change.